compiler/tests/testSuite.bal

```diff
@@ -57,15 +68,41 @@ function testCompileEU(string path, string kind) returns file:Error|io:Error? {
     }
 }
 
-function compileErrorLineNumber(CompileError err) returns int? {
-    if err is io:Error {
-        return ();
+function moduleDir(string filePath) returns string|file:Error? {
+    string subModPath = filePath.substring(0, filePath.length()-4) + ".modules";
+    if check file:test(subModPath, file:IS_DIR) {
+        return check file:normalizePath(subModPath, file:CLEAN);
+    }
+}
+
+function expectedErrorLocation(CompileError err, string path) returns [int, string]|file:Error|io:Error? {
+    string? modulePath = check moduleDir(path);
+    string? errorFileName = ();
+    int? errorLineNumber = ();
+    if modulePath is string {
+        foreach var dir in check file:readDir(modulePath) {
+            if !check file:test(dir.absPath, file:IS_DIR) {
+                continue;
+            }
+            foreach var file in check file:readDir(dir.absPath) {
+                [errorLineNumber, errorFileName] = findErrorLine(file.absPath, errorLineNumber);
+            }
+        }
+    }
+    [errorLineNumber, errorFileName] = findErrorLine(path, errorLineNumber);
+    if errorFileName is string && errorLineNumber is int {
+        return [errorLineNumber, errorFileName];
+    }
+}
+
+function findErrorLine(string filePath, int? currentErrorLine) returns [int?, string?] {
```

Comment:
Make the type be `[string, int]|io:Error?`

Reply:
fixed